File: src/content/fortran/factor.f90
program prime_factors
    implicit none
    integer :: n, i

    ! Read the number from the user
    write(*, *) "Enter a number:"
    read(*, *) n

    ! Find and print the prime factors
    write(*, *) "Prime factors of", n, ":"
    do i = 2, n
        if (mod(n, i) == 0) then
            write(*, *) i
            n = n / i
        end if
    end do

    stop
end program prime_factors

File: src/content/lua/factor.lua
-- Function to find prime factors
function primeFactors(n)
    local factors = {}
    for i = 2, n do
        while n % i == 0 do
            table.insert(factors, i)
            n = n / i
        end
    end
    return factors
end

-- Read number from user
io.write("Enter a number: ")
local number = tonumber(io.read())

-- Find and print prime factors
local factors = primeFactors(number)
io.write("Prime factors of " .. number .. ": ")
for _, factor in ipairs(factors) do
    io.write(factor .. " ")
end
print()